modules/hilary/manifests/init.pp

```diff
@@ -146,7 +146,16 @@
       owner   => $os_user,
       group   => $os_group,
       content => template('hilary/config.js.erb'),
-      require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ]
+      require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ];
+
+    # Env specific config file
+    "${app_root_dir}/${environment}.js":
+      ensure  => present,
+      mode    => "0644",
+      owner   => $os_user,
+      group   => $os_group,
+      source => "puppet:///modules/hilary/$environment.js",
```

Comment:
As $environment is used within a string it should have braces to delimit it;
  source => "puppet:///modules/hilary/${environment}.js",
Otherwise I think we're there now.